### PR TITLE
added user registration and login

### PR DESCRIPTION
**Pull request by tivvit**

### CODE REVIEW

File: server/stycoon/users.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"golang.org/x/crypto/bcrypt"
+)
+
+func HashPassword(password string) (string, error) {
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	return string(hash), err
+}
+
+func IsPasswordValid(hash []byte, pass []byte) error {
```
Comment (malytomas) [inline server/stycoon/users.go:12]: make consistent with HashPassword, string vs byte array

File: server/database/users.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func InsertUser(db *sql.DB, username string, password string) error {
+	_, err := db.Exec(`insert into d_user (name, password) values (?, ?)`, username, password)
+	return err
+}
+
+func GetUserPassword(db *sql.DB, username string) (string, error) {
+	var password sql.NullString
+	err := db.QueryRow("select password from d_user where name = ? limit 1;", username).Scan(&password)
```
Comment (malytomas) [inline server/database/users.go:15]: is the semicolon required?

File: server/handlers/users.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	var loginUser User
+	err = json.Unmarshal(body, &loginUser)
+	if err != nil {
+		log.Warn().Err(err).Msg("Json unmarshall failed")
+		http.Error(w, "can't parse json", http.StatusBadRequest)
+		return
+	}
+	hash, err := database.GetUserPassword(db, loginUser.Username)
+	if err != nil {
+		log.Warn().Err(err).Str("username", loginUser.Username).Msg("user search failed")
```
Comment (curusarn) [inline server/handlers/users.go:74]: Better message: "user not found" ?

File: server/handlers/users.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		http.Error(w, "", http.StatusBadRequest)
```
Comment (curusarn) [inline server/handlers/users.go:75]: Shouldn't this also be "StatusForbidden"?

File: server/database/users.go (hunk 2 above)
Comment (tivvit) [reply server/database/users.go:15]: no

File: server/stycoon/users.go (hunk 1 above)
Comment (tivvit) [reply server/stycoon/users.go:12]: done

File: server/handlers/users.go (hunk 4 above)
Comment (tivvit) [reply server/handlers/users.go:75]: I was trying to be clever and not return a different response for non-existent user and invalid password and I am returning different things for them, let me fix that

**Review by evamayerova — APPROVED** [review]
LGTM

**Pull request merged**